theano/gpuarray/conv_desc.c

```diff
@@ -1,25 +1,65 @@
 #section support_code_apply
 
-int APPLY_SPECIFIC(conv_desc)(PyArrayObject *filt_shp,
+int APPLY_SPECIFIC(conv_desc)(PyArrayObject *filt_shp, 
+                              PyArrayObject *padding,
+                              PyArrayObject *subsample,
+                              PyArrayObject *conv_mode,
+                              PyArrayObject *precision,
+                              PyArrayObject *bmode,
+                              PyArrayObject *nb_dims,
                               cudnnConvolutionDescriptor_t *desc) {
   cudnnStatus_t err;
-  int pad[3] = {PAD_0, PAD_1, PAD_2};
-  int strides[3] = {SUB_0, SUB_1, SUB_2};
+  int pad[3] = {*(npy_int8 *)PyArray_GETPTR1(padding, 0),
+                *(npy_int8 *)PyArray_GETPTR1(padding, 1),
+                *(npy_int8 *)PyArray_GETPTR1(padding, 2)};
+  int strides[3] = {*(npy_int8 *)PyArray_GETPTR1(subsample, 0),
+                    *(npy_int8 *)PyArray_GETPTR1(subsample, 1),
+                    *(npy_int8 *)PyArray_GETPTR1(subsample, 2)};
```

Comment:
Don't use int8 here either.

Also, the arrays may not have a third element if the inputs are 2d.